sorting/select_k/select_k.py

```diff
@@ -0,0 +1,74 @@
+def select_k(arr, k):
```

Comment:
Would be great to add asymptotic complexity (O(?)) to docstrings of non-trivial functions (e.g. `select_k`, `split_array`)

Reply:
Also would be great to give explanation general comments for non-trivial algorithms

Reply:
This seems like to have expected linear time?